src/grpc/request_template.rs

```diff
@@ -32,6 +36,13 @@ pub struct RenderedRequestTemplate {
     pub operation: ProtobufOperation,
 }
 
+impl Hash for RenderedRequestTemplate {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.url.hash(state);
+        self.body.hash(state);
+    }
+}
```

Comment:
Add unit tests for hash. Write a test where we guarantee if the url is diff the hash is diff, where the body is diff the hash is diff etc.